Get service module working with sudo, add state=status, better error messages.

jkleint opened the pull request:
When running the service module via sudo, `$PATH` wouldn't contain `/sbin`, so the service 
binary couldn't be found.  This tacks `/sbin` onto the path before running service.  It 
also includes the stdout and stderr on failure.  Added `state=status` (doesn't make the most
sense, I agree ;) that shows the output of `service <cmd> status`.

mpdehaan commented:
the minimal patch here seems to just call /sbin/service vs 'service', why not this?

mpdehaan commented:
Also state=status does not make sense, true, if you want to inquire about the status, I think you want list=status

You should probably just print the out/err results from the service on stderr so it can be debugged easily with -D.

jkleint commented:
I just don't know that every system has service located in /sbin, and we should probably respect peoples' $PATH if they want to change it for some reason.

list=status makes better sense.

I don't really know how -D works; do you mean print out/err for the status or the start/stop or both? in every case, or just the error case?  In place of or in addition to sticking it in the json?

mpdehaan commented:
In systems programming, ignoring path is generally a good thing to do, because you shouldn't trust or rely on the user environment.   If it becomes a problem we can convert the shell invocation to look in multiple places.

mpdehaan commented:
basically -D means anything (unformatted, even) the module spews on stderr will show up when running Ansible.

so when you run the service command, capture stdout/stderr, and just print it back on stderr (both of them), and don't include it in the structured JSON output.

Keeps the amount of stuff that Ansible is keeping around in memory or returning small, also.

jkleint commented:
OK, I've hardcoded /sbin/service, changed to list=status, and print service's stdout + stderr to stderr on error.  Let me know if that's what you meant.

Commit:
Get service module working with sudo, add list=status, better error messages.

When running the service module via sudo, `$PATH` didn't contain `/sbin`,
so the service binary couldn't be found.  This just runs `/sbin/service`
directly.  Output is spewed to stderr on error.

Added `list=status` to include the output of `service <cmd> status`.

mpdehaan commented:
sounds reasonable, will test shortly, thanks!
